eng/common/scripts/Verify-Links.ps1

```diff
@@ -269,23 +284,38 @@ while ($pageUrisToCheck.Count -ne 0)
 
   $linkUris = GetLinks $pageUri
   Write-Host "Found $($linkUris.Count) links on page $pageUri";
-  
+  $badLinksPerPage = @();
   foreach ($linkUri in $linkUris) {
+    Write-Host "before the replace $linkUri"
+    Write-Host "the commit $branchReplacementName"
     $linkUri = ReplaceGithubLink $linkUri
-
+    Write-Host "after the replace $linkUri"
     $isLinkValid = CheckLink $linkUri
-    if (!$isLinkValid) {
-      $script:badLinks += $linkUri
+    if (!$isLinkValid -and !$badLinksPerPage.Contains($linkUri)) {
+      $badLinksPerPage += $linkUri
     }
     if ($recursive -and $isLinkValid) {
       if ($linkUri.ToString().StartsWith($baseUrl) -and !$checkedPages.ContainsKey($linkUri)) {
         $pageUrisToCheck.Enqueue($linkUri);
       }
     }
   }
+  if ($badLinksPerPage.Count -gt 0) {
+    $badLinks[$pageUri] = $badLinksPerPage
+  }
 }
 
-Write-Host "Found $($checkedLinks.Count) links with $($badLinks.Count) broken"
-$badLinks | ForEach-Object { Write-Host "  $_" }
+foreach ($pageLink in $badLinks.Keys) {
+  Write-Host "  '$pageLink' has $($badLinks[$pageLink].Count) broken link(s)`:`n"
```

Comment:
```suggestion
  Write-Host "'$pageLink' has $($badLinks[$pageLink].Count) broken link(s)`:"
```

Reply:
The space here and the following is for well reading.
```
Summery of broken links:
  "reamd me link" has 2 broken links:
     "broken link 1"
     "broken link 2"
     ...
```

I suggest to leave as it is.